File: src/main/java/com/uapp_llc/model/Column.java
package com.uapp_llc.model;

import java.util.ArrayList;
import java.util.List;
import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.OrderBy;
import javax.persistence.Table;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "columns")
public class Column {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @javax.persistence.Column(name = "id")
  private Long id;

  @javax.persistence.Column(name = "name")
  private String name;

  @javax.persistence.Column(name = "index", nullable = false)
  private Integer index;

  @OneToMany(mappedBy = "column", cascade = CascadeType.REMOVE)
  @OrderBy("index")
  private List<Task> tasks = new ArrayList<>();

}

File: src/main/kotlin/com/uapp_llc/model/Column.kt
package com.uapp_llc.model

import javax.persistence.CascadeType
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.OrderBy
import javax.persistence.Table

@Entity
@Table(name = "columns")
class Column(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @javax.persistence.Column(name = "id")
    var id: Long? = null,

    @javax.persistence.Column(name = "name")
    var name: String? = null,

    @javax.persistence.Column(name = "index", nullable = false)
    var index: Int

) {

  @OneToMany(mappedBy = "column", cascade = [CascadeType.REMOVE])
  @OrderBy("index")
  var tasks: MutableList<Task> = mutableListOf()

}
